ocis-reva/docs/testing.md

```diff
@@ -0,0 +1,148 @@
+```
+
+Then you need to start the ocis-reva services
+```
+bin/ocis-reva frontend & \
+bin/ocis-reva gateway & \
+bin/ocis-reva auth-basic & \
+bin/ocis-reva auth-bearer & \
+bin/ocis-reva sharing & \
+bin/ocis-reva storage-home & \
+bin/ocis-reva storage-home-data & \
+bin/ocis-reva storage-oc & \
+bin/ocis-reva storage-oc-data & \
+bin/ocis-reva users &
+```
+
+### Run the API acceptance tests
+
+In the ownCloud 10 core repo run
+
+```
+make test-acceptance-api \
+TEST_SERVER_URL=http://localhost:9140 \
+TEST_EXTERNAL_USER_BACKENDS=true \
+TEST_OCIS=true \
+OCIS_REVA_DATA_ROOT=/var/tmp/reva/ \
+BEHAT_FILTER_TAGS='~@notToImplementOnOCIS&&~@toImplementOnOCIS&&~@preview-extension-required' \
+SKELETON_DIR=apps/testing/data/apiSkeleton
+```
+
+Make sure to adjust the settings `TEST_SERVER_URL`,`OCIS_REVA_DATA_ROOT` and `SKELETON_DIR` according to your environment.
+
+This will run all tests that are relevant to OCIS.
+
+To run a single test add `BEHAT_FEATURE=<feature file>` and specify the path to the feature file and an optional line number. For example: `BEHAT_FEATURE='tests/acceptance/features/apiWebdavUpload1/uploadFile.feature:12'`
+
+### use existing tests for BDD
+
+As a lot of scenarios are written for oC10, we can use those tests for Behaviour driven development in ocis.
+Every scenario that does not work in OCIS with OC storage, is listed in `tests/acceptance/expected-failures-on-OC-storage.txt` with a link to the related issue.
+Similarly, scenarios that do not work in OCIS with EOS storage are listed in `tests/acceptance/expected-failures-on-EOS-storage.txt`.
+Scenarios from the oC10 API acceptance tests are run in the ordinary acceptance test pipeline in CI. The scenarios that fail are checked against the
+expected failures. If there are any differences then the CI pipeline fails.
+
+Additionally, some issues have scenarios that demonstrate the current buggy behaviour in ocis(reva).
+Those scenarios are in this ocis-reva repository in `tests/acceptance/features/apiOcisSpecific`.
+Have a look into the [documentation](https://doc.owncloud.com/server/developer_manual/testing/acceptance-tests.html#writing-scenarios-for-bugs) to understand why we are writing those tests.
+Also, ocis behaves partly differently with EOS-Storage and OC-Storage. There are scenarios that do not work in OCIS when run on EOS-storage, but works when on OC-Storage, and vice-versa. For those kind of scenarios, ` @skipOnOcis-EOS-Storage` and `@skipOnOcis-OC-Storage` tags are used. For instance, for a scenario that fails on EOS-Storage but passes on OC-Storage, we use `@skipOnOcis-EOS-Storage` tag to let it run on OC-Storage, where it works as expected, instead of skipping the test completely.
+
+If you want to work on a specific issue
+
+1.  adjust the core commit id to the latest commit in core so that CI will run the latest test code and scenarios from core.
+    For that change `coreCommit` in the `config` section:
+
+        config = {
```

Comment:
Codacy found an issue: [[code-block-style] Code blocks should be fenced](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6202733)